### PR TITLE
[SYCL][Matrix] Move/Refactor matrix check_device_code tests to use SYCL_EXTERNAL

### PR DESCRIPTION
**Pull request by ianayl**
In the same vein as #14174, this PR:
- Moves matrix device code checks that are not in the `check_device_code` folder into the `check_device_code/matrix` folder
- Removed host code from tests in `sycl/test/check_device_code/matrix` that check only for device code using `SYCL_EXTERNAL` functions (as per #14173)

### CODE REVIEW

File: sycl/test/check_device_code/matrix/matrix_load_store_as.cpp [hunk 1]
```diff
@@ -9,55 +9,32 @@
...
+  joint_matrix_load(
+      sg, tA, tileA.template get_multi_ptr<sycl::access::decorated::yes>(), 16);
+  // B should load from global address space
+  // CHECK: %{{.*}} = tail call spir_func noundef target("spirv.JointMatrixINTEL", i16, 16, 16, 2, 3, 1) @_Z[[#]]__spirv_JointMatrixLoadINTEL{{.*}}(ptr addrspace(1) noundef %{{.*}}, i64 noundef 32, i32 noundef 2, i32 noundef 3, i32 noundef 0) #{{.*}}
+  joint_matrix_load(sg, tB, pB, 32);
+  joint_matrix_mad(sg, tC, tA, tB, tC);
+  // C should store to global address space
+  // CHECK: tail call spir_func void @_Z[[#]]__spirv_JointMatrixStoreINTEL{{.*}}(ptr addrspace(1) noundef %{{.*}}, target("spirv.JointMatrixINTEL", float, 8, 16, 3, 3, 2) noundef %{{.*}}, i64 noundef 16, i32 noundef 0, i32 noundef 3, i32 noundef 0) #{{.*}}
+  joint_matrix_store(sg, tC, pC, 16, layout::row_major);
+}
```
Comment (YuriPlyakhin) [inline sycl/test/check_device_code/matrix/matrix_load_store_as.cpp:40]: nit: add new line at end of file.

**Review by YuriPlyakhin — APPROVED** [review]

File: sycl/test/check_device_code/matrix/matrix_load_store_as.cpp (hunk 1 above)
Comment (ianayl) [reply sycl/test/check_device_code/matrix/matrix_load_store_as.cpp]: Fixed, thanks for the catch!

**Comment by ianayl** [comment]
@YuriPlyakhin Please excuse the ping, do I contact you for CI/workflow approval? Thanks!

**Comment by ianayl** [comment]
@YuriPlyakhin Please excuse the ping, could you look over the new changes here before I ask gatekeepers to merge? Thanks!

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

**Comment by ianayl** [comment]
@intel/llvm-gatekeepers This PR is ready, thanks!

**Pull request merged**